### ISSUE TITLE
Merge

### ISSUE THREAD

**Issue by dnhkng** [sirluk/pytorch_incremental_pca#1]
Should I merge your code into PCAonGPU?

You have made significant changes, and it would make sense not to have my old code up if this is numerically better!

**Comment by sirluk** [comment]
@dnhkng thanks for opening the issue! If you would like I could open a PR there, its just that I made some significant changes to the api as well (init arguments and such) I dont know if you would be okay with that.

I think it could benefit users on the one hand due to the precision improvements (I reimplemented the incremental mean and variance algorithm used in sklearn) as well as the option to use truncated SVD which is significantly faster.

**Comment by dnhkng** [comment]
!00% agree, I saw you made significant changes, and complex ones at that!  But I would welcome a PR :)

If your version is superior, it should be the ones that people use. Happy to fully document that the code is yours of course.  

Maybe we could add in a table, showing the accuracy of my original code, and the improvements yours makes?

**Comment by sirluk** [comment]
Yes thats a good idea! I will see if I find time the next days to set up a small script with some toy dataset to showcase the precision difference.
